src/main/java/com/conveyal/r5/streets/StreetRouter.java

```diff
@@ -968,8 +967,8 @@ public void incrementTimeInSeconds(long seconds) {
                 durationSeconds-=seconds;
                 durationFromOriginSeconds -= seconds;
             } else {
-                durationSeconds+=seconds;
-                durationFromOriginSeconds += seconds;
+                durationSeconds += (int) seconds;
```

Comment:
Of course there doesn't seem to be any harm in explicitly performing a cast, but what are the security implications of explicitly casting to an int here, and why would we do it when adding but not when subtracting? It seems like the code scanning may latch on to certain details that just match a simple pattern implicated in a known security problem without truly validating whether the problem is present. It's still probably worthwhile to try out CodeQL, but if over the longer term it doesn't seem to target and explain true security problems I wouldn't want to keep using it just out of inertia.

On further thought I'm actually kind of wary of accepting its suggestions. The difference between the subtraction and addition branches in what the tool considers "more secure" code creates a pattern that a future reader may stop and try to interpret - why is it this way, what is it trying to achieve by explicitly casting only in one branch? It's arguable that the automated system is requiring us to degrade the maintainability of the code for no benefit.